Assets/Scripts/Spawning/ResourceSpawning/ResourceSpawner.cs

```diff
@@ -0,0 +1,88 @@
+using Examen.Spawning.ResourceSpawning.Structs;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Examen.Spawning.ResourceSpawning
+{
+    public class ResourceSpawner : MonoBehaviour
+    {
+        [SerializeField] private List<ResourceSpawnAreas> _spawnAreas = new();
+
+        private float _spawnPercentage;
+        private float _currentSpawnAmount;
+
+        private const float MAX_PERCENTAGE = 100;
+
+        private readonly List<GameObject> _spawnedGameobjects = new();
+
+        /// <summary>
+        /// Spawns randomly resources
+        /// </summary>
+        public void InitializedSpawning()
+        {
+            for (int i = 0; i < _spawnAreas.Count; i++)
+            {
+                ResourceSpawnAreas area = _spawnAreas[i];
+                _spawnPercentage = MAX_PERCENTAGE;
+                _currentSpawnAmount = area.ResourceAmount;
+
+                for (int j = 0; j < area.SpawnableResources.Count; j++)
+                {
+                    if (_spawnPercentage <= 0 || _currentSpawnAmount <= 0)
+                    {
+                        Debug.LogError($"Can't spawn {area.SpawnableResources[j].SpawnResource.name}, " +
+                            $"because you can't spawn more then 100% Resources");
+                        break;
+                    }
+
+                    float amountOfResources = CalculateResrouceAmount(area, area.SpawnableResources[j]);
```

Comment:
```suggestion
                    float resrouceAmount= CalculateResrouceAmount(area, area.SpawnableResources[j]);
```
Could you try to use less of the work `of` in variable names, usually it can be rewritten to be shorter and more clear